File: internal/model/category.go
package model

import "time"

type Category struct {
	ID        uint64    `gorm:"column:id;primaryKey;autoIncrement"`
	ParentID  uint64    `gorm:"column:parent_id"`
	Name      string    `gorm:"column:name"`
	Icon      string    `gorm:"column:icon"`
	SortOrder int       `gorm:"column:sort_order"`
	Status    int       `gorm:"column:status"`
	CreatedAt time.Time `gorm:"column:created_at"`
	UpdatedAt time.Time `gorm:"column:updated_at"`
}

func (Category) TableName() string {
	return "categories"
}

File: seller-app/src/types/category.ts
export interface SellerCategoryTreeNode {
  id: number;
  parent_id: number;
  name: string;
  icon: string;
  sort_order: number;
  status: number;
  children: SellerCategoryTreeNode[];
}

export interface SellerCategoryOption {
  id: number;
  label: string;
  parent_label?: string;
}
